pkg/db/src/implementations/version-1.2/customer-orders.ts

```diff
@@ -0,0 +1,168 @@
+			share({ connector: () => updateSubject, resetOnError: false, resetOnComplete: false, resetOnRefCountZero: false })
+		);
+		this.#stream = this.#updateStream.pipe(
+			// We're connecting the stream to a ReplaySubject as a multicast object: this enables us to share the internal stream
+			// with the exposed streams (orderId??) and to cache the last value emitted by the stream: so that each subscriber to the stream
+			// will get the 'initialValue' (repeated value from the latest stream).
+			share({ connector: () => cache, resetOnError: false, resetOnComplete: false, resetOnRefCountZero: false })
+		);
+
+		// The first value from the stream will be either customer order data, or an empty object (if the customer order doesn't exist in the db).
+		// This is enough to signal that the customer order intsance is initialised.
+		firstValueFrom(this.#stream).then(() => this.#initialized.next(true));
+		// If data is not empty (customer order exists), setting of 'exists' flag is handled inside the 'updateInstance' method.
+		this.#stream.subscribe((w) => this.updateInstance(w));
+		return this;
+	}
+
+
+	/**
+	 * If order doesn't exist in the db, create an order with initial values.
+	 * No-op otherwise.
+	 */
+	create(): Promise<CustomerOrderInterface> {
+		return runAfterCondition(async () => {
+			if (this.#exists) {
+				return this;
+			}
+
+			const updatedAt = new Date().toISOString();
+
+			const sequentialNumber = (await this.#db._pouch.query("v1_sequence/customer-order")).rows[0];
+			const seqIndex = sequentialNumber ? sequentialNumber.value.max && ` (${sequentialNumber.value.max + 1})` : "";
+
+			const initialValues = { ...this, displayName: `Order-${seqIndex}`, updatedAt };
+			const { rev } = await this.#db._pouch.put<CustomerOrderData>(initialValues);
+
+			return this.updateInstance({ ...initialValues, _rev: rev });
+		}, this.#initialized);
+	}
+
+
+	/**
+	 * Update field is a method for internal usage, used to update the local field, if the value is provided
+	 */
+	private updateField<K extends keyof CustomerOrderData>(field: K, value?: CustomerOrderData[K]) {
+		if (value !== undefined) {
+			this[field] = value as any;
+		}
+		this.#exists = true;
+		return this;
+	}
+	/**
+	 * Update instance is a method for internal usage, used to update the instance with the data (doesn't update the DB)
+	 */
+	private updateInstance(data: Partial<Omit<CustomerOrderData, "_id">>): CustomerOrderInterface {
+		// No-op if the data is empty
+		if (isEmpty(data)) {
+			return this;
+		}
+
+		// Update the data with provided fields
+		this.updateField("_rev", data._rev);
+		this.updateField("orderId", data.orderId);
+		this.updateField("books", data.books);
+		this.updateField("deposit", data.deposit);
+		this.updateField("docType", data.docType);
+		this.updateField("updatedAt", data.updatedAt);
+		this.updateField("email", data.email);
+
+		this.#exists = true;
+
+		return this;
+	}
```

Comment:
Can we make it so that the class definition follows this structure:
- declaration of private properties (properties prepended with `#`)
- declaration of data (non-private properties)
- constructor (if any)
- private methods
- public methods